### PR TITLE
Add subscribe method to the StreamLayerClient

### PR DESCRIPTION
**Pull request by AndriyShevchuk**
Subscribe method allows user to create a subscription to the stream layer and get base URL to get data from the layer.
* Add subscribe method to the StreamLayerClient
* Add unit tests

Resolves: OLPEDGE-1522
Signed-off-by: ashevchu <ext-andrii.shevchuk@here.com>

### CODE REVIEW

File: @here/olp-sdk-dataservice-read/lib/DataStoreDownloadManager.ts [hunk 1]
```diff
@@ -96,7 +96,8 @@ export class DataStoreDownloadManager implements DownloadManager {
             const response = await fetchFunction(url, init);
             // tslint:disable-next-line: no-magic-numbers
             if (response.status !== 503 || retryCount >= maxRetries) {
-                if (response.status === 200) {
+                // tslint:disable-next-line: no-magic-numbers
```
Comment (OleksiiZubko) [inline @here/olp-sdk-dataservice-read/lib/DataStoreDownloadManager.ts]: Please create  enum with status codes and remove ` // tslint:disable-next-line: no-magic-numbers`

File: @here/olp-sdk-dataservice-read/lib/DataStoreRequestBuilder.ts [hunk 2]
```diff
@@ -32,10 +32,7 @@ async function addBearerToken(
      * @todo fix datastore-api template generator.
      * It sends empty object for headers, but should be Headers object
      */
-    if (
-        init.headers === undefined ||
-        Object.entries(init.headers).length === 0
-    ) {
+    if (init.headers === undefined || !(init.headers instanceof Headers)) {
```
Comment (OleksiiZubko) [inline @here/olp-sdk-dataservice-read/lib/DataStoreRequestBuilder.ts]: what about if init.headers will be an empty object?

Comment (AndriyShevchuk) [reply @here/olp-sdk-dataservice-read/lib/DataStoreRequestBuilder.ts]: Than we'll replace it with Headers instance. Nothing will be lose

File: @here/olp-sdk-dataservice-read/lib/DataStoreDownloadManager.ts (hunk 1 above)
Comment (AndriyShevchuk) [reply @here/olp-sdk-dataservice-read/lib/DataStoreDownloadManager.ts]: Done

**Pull request merged**